sdk/eventhub/azure-eventhubs/azure/eventhub/aio/producer_async.py

```diff
@@ -110,39 +110,30 @@ def _create_handler(self):
             self.target = self.redirected.address
         await super(EventHubProducer, self)._open(timeout_time)
 
-    async def _send_event_data(self, timeout=None):
-        timeout = timeout or self.client.config.send_timeout
-        if not timeout:
-            timeout = 100000  # timeout None or 0 mean no timeout. 100000 seconds is equivalent to no timeout
-        start_time = time.time()
-        timeout_time = start_time + timeout
-        max_retries = self.client.config.max_retries
-        retry_count = 0
-        last_exception = None
-        while True:
-            try:
-                if self.unsent_events:
-                    await self._open(timeout_time)
-                    remaining_time = timeout_time - time.time()
-                    if remaining_time < 0.0:
-                        if last_exception:
-                            error = last_exception
-                        else:
-                            error = OperationTimeoutError("send operation timed out")
-                        log.info("%r send operation timed out. (%r)", self.name, error)
-                        raise error
-                    self._handler._msg_timeout = remaining_time  # pylint: disable=protected-access
-                    self._handler.queue_message(*self.unsent_events)
-                    await self._handler.wait_async()
-                    self.unsent_events = self._handler.pending_messages
-                    if self._outcome != constants.MessageSendResult.Ok:
-                        if self._outcome == constants.MessageSendResult.Timeout:
-                            self._condition = OperationTimeoutError("send operation timed out")
-                        _error(self._outcome, self._condition)
-                return
-            except Exception as exception:
-                last_exception = await self._handle_exception(exception, retry_count, max_retries, timeout_time)
-                retry_count += 1
+    @_retry_decorator
+    async def _send_event_data(self, **kwargs):
```

Comment:
You should take `*args` as the first parameter, or else positional parameters will not be passed through/the decorator won't work.